plugins/pagerduty/src/components/Incident/Incidents.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useEffect } from 'react';
+import { List, ListSubheader, LinearProgress } from '@material-ui/core';
+import { IncidentListItem } from './IncidentListItem';
+import { IncidentsEmptyState } from './IncidentEmptyState';
+import { useAsyncFn } from 'react-use';
+import { pagerDutyApiRef } from '../../api';
+import { useApi } from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+
+type Props = {
+  serviceId: string;
+  shouldRefreshIncidents: boolean;
+  setShouldRefreshIncidents: (shouldRefreshIncidents: boolean) => void;
+};
+
+export const Incidents = ({
+  serviceId,
+  shouldRefreshIncidents,
+  setShouldRefreshIncidents,
```

Comment:
This interface looks odd to me and is probably risky - you aren't supposed to setState from external places and outside of the render loop. If this component wants to inform the outside world that it wants a refresh of something, it should just take an `onTriggerRefresh: () => void` parameter. If it just wants to drive refreshes internally, the use of useAsyncFn below should be enough.

If you do want do drive the data lifecycle from the outside, you could also have a wrapping data provider component that does all the loading and can retrigger etc, and then make these components purely visual and props driven instead.